job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -561,9 +562,18 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
-    //no authentication required
     get { ctx =>
-      ctx.complete("OK")
+      try {
+          val healthCheckClass = Class.forName(config.getString("spark.jobserver.healthcheck"))
```

Comment:
Health check is something that is called very often, wouldn't it be better to create a global variable in this class and instantiate the healthcheck object once?

Reply:
Would be possible to create HealthCheck class still in Jobserver.scala and pass it to WebApi?
Then there is no need to pass WebApi daoActor, which is a dependency of HealthCheckService.

Reply:
@valan4ik I get the point that I can create HealthCheck class in JobServer.scala itself and pass to WebApi. But I do not understand why there is no need to pass daoActor. While invoking the healthCheckImpl function from WebApi, I might still need to pass daoActor right?

Reply:
ping @valan4ik

Reply:
From my understanding it should look like this:
- Create helthCheckImpl instance in JobServer.scala with all needed parameters: daoActor and any other actor we may add later.
- Pass already created object to WebApi. WebApi doesn't need daoActor as far as I remember. All calls to daoActor happen inside healthCheckImpl, so you just call needed methods of healthCheckImpl.

A huge advantage is that if healthCheckImpl takes any additional parameters later, there is no need to add new parameters for WebApi. Only change the initialization of the object in JobServer.scala.

Hope I didn't miss anything during review :)

Reply:
Thank you. I got the point. I will construct the object in JobServer.scala.

Reply:
Okay, I looked into the code again. It's maybe a bit following @bsikander comment as well. 
At the moment you create a new instance here each time. But there is no need for it.
supervisor, jobInfoActor, daoActor don't change.
It's enough to initialize class with this parameters:
```
class ActorsHealthCheck(ctx: RequestContext, supervisor: ActorRef, jobInfo: ActorRef, daoActor: ActorRef) 
extends HealthCheckService
...
```
save them as class variables and just call `healthCheckImpl()` function on a class, without passing `supervisor, jobInfoActor, daoActor` each time.

Meaning depending on implementation class which extends `HealthCheckService` may take different parameters, but main function `healthCheckImpl` may be renamed just to `executeHealthCheck` and don't take any parameters.
Then we just need to be careful during the initialization of the health check instance.

Not sure if @bsikander had smth similar in mind.

Reply:
Addressed